src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodec.java

```diff
@@ -167,10 +168,49 @@ static <JavaT extends Number, CqlT> ToCQL<JavaT, CqlT> safeNumber(
         }
       };
     }
+
+    static Integer safeLongToInt(Long value) {
+      long l = value.longValue();
+      if (l < Integer.MIN_VALUE) {
+        throwUnderflow(DataTypes.INT, value);
+      } else if (l > Integer.MAX_VALUE) {
+        throwOverflow(DataTypes.INT, value);
+      }
+      return (int) l;
+    }
+
+    static Short safeLongToSmallint(Long value) {
+      long l = value.longValue();
+      if (l < Short.MIN_VALUE) {
+        throwUnderflow(DataTypes.SMALLINT, value);
+      } else if (l > Short.MAX_VALUE) {
+        throwOverflow(DataTypes.SMALLINT, value);
+      }
+      return (short) l;
+    }
+
+    static Byte safeLongToTinyint(Long value) {
+      long l = value.longValue();
+      if (l < Byte.MIN_VALUE) {
+        throwUnderflow(DataTypes.TINYINT, value);
+      } else if (l > Byte.MAX_VALUE) {
+        throwOverflow(DataTypes.TINYINT, value);
+      }
+      return (byte) l;
+    }
+
+    static void throwOverflow(DataType targetCQLType, Number value) {
+      throw new ArithmeticException(String.format("Overflow, value too big for %s", targetCQLType));
```

Comment:
Good point -- I was doing that, but realized that `safeNumber()` that catches `ArithmeticException` and re-wraps as `ToCQLCodecException` adds the actual value.
This is why I chose to avoid adding it in `ArithmeticException` itself.

On underflow: yeah I must admit I googled for definitions since I recall "overflow" sometimes being used to mean both, too. :)